### PR TITLE
test: big cleanup followup + typing

### PR DESCRIPTION
**Pull request by lewis6991**
Follow up to #26994

- Fix final uses of the API that didn't include the `nvim_` prefix
- do not merge the `test.cmakeconfig.paths` into the `helpers` and instead add it as a `helpers.paths`
- Remove some unused functions:
  - `helpers.is_arch()`
  - `helpers.hasenv()`
  - `helpers.updated()`
  - `helpers.fixtbl()` and `helpers.fixtbl_rec()`
  - `helpers.isfile()`
- remove the `test.deprecated` module
- Moved `helpers.format_{string,luav}()` to a separate module
  - also replaced `helpers.REMOVE_THIS` with `vim.NIL` to use as a sentinel value in some functions.
- Added lots of type annotations.

### CODE REVIEW

File: runtime/lua/vim/inspect.lua [hunk 1]
```diff
@@ -1,3 +1,9 @@
+--- @class vim.inspect.Opts
+--- @field depth? integer
+--- @field newline? string
+--- @field process? fun(item:any, path: string[]): any
+
+--- @overload fun(x: any, opts?: vim.inspect.Opts): string
```
Comment (wookayin) [inline runtime/lua/vim/inspect.lua:6]: Nice, I also once needed this! But in order to make this bundled file as-is (although we already have some neovim-specific modifications), how about adding the annotations to `vim/_editor.lua`?

L199: `vim.inspect = vim.inspect`

Comment (lewis6991) [reply runtime/lua/vim/inspect.lua:6]: We have quite a few Neovim specific modifications to inspect.lua so I think we basically own that file now.

I can move the annotations anyway.

Comment (clason) [reply runtime/lua/vim/inspect.lua:6]: Yes, upstream is in teal (🤢) and bfredl made a number of changes to support inspecting Vim types (like `vim.NIL`). All our inspect belong to us now.

Comment (lewis6991) [reply runtime/lua/vim/inspect.lua]: done

**Pull request merged**